### DISCUSSION TITLE
Is it possible to configure this to use exclusively modules and no commonjs?

### DISCUSSION THREAD

**Discussion by mapleroyal** [Deluze/electron-vue-template#67] [Q&A] [answered]
I can figure out setting `"type": "module",` in package.json, and in `src/main/main.tsconfig` changing `commonjs` to `esnext` and adding `"moduleResolution": "node"`, as well as updating the vite.config.js to modern syntax. But things start breaking after that and it's not clear what else to update.

**Comment by Deluze** [comment, marked as answer]
What version of Electron are you on?

ESM support was added in Electron 28, this template is still on Electron 27. You should be able to update this version with ease.
There are some things to keep in mind though, Electron has a documentation page regarding ESM support: https://www.electronjs.org/docs/latest/tutorial/esm

**Comment by mapleroyal** [comment]
I was just using the template as is, so Electron 27. I had no awareness that modules support is new, let alone incomplete, in Electron.

Thanks for clarifying this!
